packages/http/src/mocker/negotiator/InternalHelpers.ts

```diff
@@ -34,59 +34,56 @@ export function findBestHttpContentByMediaType(
 
   return pipe(
     response.contents,
-    findFirst(content => content.mediaType === bestType),
+    findFirst(content => content.mediaType === bestType)
   );
 }
 
 export function findDefaultContentType(
-  response: PickRequired<IHttpOperationResponse, 'contents'>,
+  response: PickRequired<IHttpOperationResponse, 'contents'>
 ): Option<IMediaTypeContent> {
   return pipe(
     response.contents,
-    findFirst(content => content.mediaType === '*/*'),
+    findFirst(content => content.mediaType === '*/*')
   );
 }
 
 const byResponseCode = ord.contramap<number, IHttpOperationResponse>(ordNumber, response => parseInt(response.code));
 
-export function findLowest2xx(httpResponses: IHttpOperationResponse[]): Option<IHttpOperationResponse> {
+export function findLowest2xx(httpResponses: NonEmptyArray<IHttpOperationResponse>): Option<IHttpOperationResponse> {
```

Comment:
I'm aware of the discussion on http-spec typings. Just thinking of using IHttpOperation['responses'] as a type here to be consistent with what we could receive. Despite the fact of how bad the input things can be.

Reply:
I would say these function will take a non empty array, we will validate this ahead of the time, most likely in the CLI.

If we're going to receive garbage input — we're going to make sure to validate it before it enters in Prism-Http.

Reply:
Sounds good!

Reply:
Well, almost good. I would be happier when we could rely on IHttpOperation for that.